v3/lints/cabf_ev/lint_extra_subject_attribs.go

```diff
@@ -0,0 +1,100 @@
+/*
+ * ZLint Copyright 2024 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/*
+ * Contributed by Adriano Santoni <asantoni64@gmail.com>
+ */
+
+package cabf_ev
+
+import (
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/v3/lint"
+	"github.com/zmap/zlint/v3/util"
+
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"fmt"
+)
+
+func init() {
+	lint.RegisterCertificateLint(&lint.CertificateLint{
+		LintMetadata: lint.LintMetadata{
+			Name:          "e_ev_extra_subject_attribs",
+			Description:   "CAs SHALL NOT include any Subject Distinguished Name attributes except as specified...",
+			Citation:      "EVGs §7.1.4.2.9",
+			Source:        lint.CABFEVGuidelines,
+			EffectiveDate: util.SC16EffectiveDate,
+		},
+		Lint: NewExtraSubjectAttribs,
+	})
+}
+
+type extraSubjectAttribs struct{}
+
+func NewExtraSubjectAttribs() lint.LintInterface {
+	return &extraSubjectAttribs{}
+}
+
+func (l *extraSubjectAttribs) CheckApplies(c *x509.Certificate) bool {
+	return util.IsEV(c.PolicyIdentifiers) && util.IsSubscriberCert(c)
+}
+
+/*
+ * We also include the OU attribute here, even though it is now banned, because this lint
+ * deals with a more general requirement that came into force long before the OU ban,
+ * and there is already another lint that deals with the OU attribute specifically.
+ *
+ * The organizationIdentifier attribute is only permitted starting from 21-may-2019 (EVGL 1.7.0),
+ * which is slightly after SC16 came into force, however any certificates that contain this
+ * attribute and were issued before that date have long since expired, so it makes no difference.
+ */
+var allowedAttribs = map[string]bool{
+	"1.3.6.1.4.1.311.60.2.1.1": true, // joiLocalityName
```

Comment:
Thank you for taking the time to clearly enumerate these. It must have been at least mildly tedious.